### ISSUE TITLE
Show grid lines 

### ISSUE THREAD

**Issue by ronvanpol** [gridstack/gridstack.js#662]
Does anyone have any idea of how to add grid lines to a gridstack? When working with bigger gridstacks, it helps a lot to place an item on the gridstack if grid lines could be shown (for instance like Excel)

**Comment by radiolips** [comment]
Good question, @ronvanpol . You'll want to do this with CSS. Maybe someone better than I am at CSS will be able to offer you some assistance. If not, I'll spend some time looking into it when I have a chance. Thanks for using gridstack! You may want to check out the [gridstack Slack](https://gridstackjs.troolee.com/) to get a wider audience's attention as well

**Comment by sohaib-khan** [comment]
Hi @radiolips. Is there any update on showing grid lines lets say if a grid is initialized for 12 col and 12 rows to begin with?

**Comment by radiolips** [comment]
Hey @sohaib-khan . We haven't had a chance to look into that just yet. We're hard at work with some bugs and major [1.0 features](https://dylandreams.com/2017/04/26/gridstack-10-coming-soon/). When I have a chance to look into these grid lines, I'll update this ticket. Thanks for the ping. I'd love to one day make gridstack.js a full-time job, but for the time being, I'm devoting as much time as I can :).

**Comment by sohaib-khan** [comment]
@radiolips great! I will be following when this new feature kicks in. It will be a good visual cue for snapping divs to the grid.

**Comment by Prashik123** [comment]
Hi we can try following css workaround for now(try manipulating background-size based on ur requirement):

    .grid-stack {
       /* background-color: #eeeeee; */
    background-image: linear-gradient(#e0e0e0 1px, #00000000 1px), linear-gradient(90deg, #e0e0e0 1px, transparent 0px), linear-gradient(rgba(255,255,255,.3) 5px, transparent 100px), linear-gradient(90deg, rgba(255,255,255,.3) 5px, transparent 100px);
    background-size: 100px 62px, calc(8.33% + 0px) 100px, 20px 20px, 20px 20px;
    background-position: -2px -2px, -1px -2px, -1px -1px, -1px -1px;
}  
![image](https://user-images.githubusercontent.com/4114326/37252936-98015eee-2550-11e8-9d38-c56a0bfe6b0a.png)

**Comment by dfenerski** [comment]
is there a way to combine the solution mentioned above with usage of custom number of columns?

**Comment by tomasrivero** [comment]
Yes, the part where it says 'calc(8.33% + 0px)' replace it with 'calc(100/N% + 0px)' where N is your amount of columns

**Comment by dfenerski** [comment]
@tomasrivero this seems good, but what about the case of custom number of rows

**Comment by Justin99b** [comment]
Any planes yet when you would implement this if you will ever implement it?

**Comment by adumesny** [comment]
@Justin99b I wasn't planning on (low on the todo list) unless someone wants to donate to make it happen

**Comment by liesahead** [comment]
Doesn't this css fit?

```scss
    background-color: #F7F7F7;
    background-size : $cell-size $cell-size;
    background-image:
        linear-gradient(to right, $border-color 1px, transparent 1px),
        linear-gradient(to bottom, $border-color 1px, transparent 1px);
```

**Comment by atlemagnussen** [comment]
I have used @liesahead 's example and make it work

```css
background-color: grey;
    background-size: calc(100%/40) calc(100%/var(--gs-current-row, 30));
    background-image:
        linear-gradient(to right, #FFF 1px, transparent 1px),
        linear-gradient(to bottom, #fff 1px, transparent 1px);
```

I use javascript to get the attributte `gs-current-row` from the grid element and then set it as a CSS Custom Property in inline style, so the CSS can use it for calc

However, for dynamic grid that will expand and shrink between min and max rows, I can't find a way to detect when the grid size changes

**Comment by delvinwidjaja** [comment]
hi everyone, i've been trying to adjust the grid line so it would look like this but havent found any solution (yet):
<img width="296" alt="Screenshot 2024-04-24 at 15 30 06" src="https://github.com/gridstack/gridstack.js/assets/44110031/7d6e15a0-b6f7-46cf-98d3-3727cd626c59">

any suggestion is appreciated. thanks!

**Comment by adrianbenjuya** [comment]
I share my solution for inspiration based on @liesahead and @atlemagnussen examples:

The CSS
```css
gridstack {
   background-size: calc(100% / var(--columns)) var(--cell-height);
   background-image: linear-gradient(to right, #fff 1px, transparent 1px),
      linear-gradient(to bottom, #fff 1px, transparent 1px);
}
```

The Angular code
```typescript
@Component({...})
export class TheComponent {
   @HostBinding('style.--columns')
   columns = grid.getColumns();

   @HostBinding('style.--cell-height')
    cellHeight = grid.getCellHeight + 'px';
}
```

**Comment by adumesny** [comment]
@adrianbenjuya cool. always learning something new, binding angular var to CSS style in this case.

**Comment by ray007** [comment]
I found https://stackoverflow.com/questions/3540194/how-to-make-a-grid-like-graph-paper-grid-with-just-css helpful for this.
